app/main.py

```diff
@@ -1,25 +1,94 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(
+            self,
+            instance: int,
+            owner: IntegerRange
+    ) -> None:
+        getattr(instance, self.name)
+
+    def __set__(
+            self,
+            instance: int,
+            value: int
+    ) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.name, value)
```

Comment:
You should raise an exception in case the check is not successful. Later you could define whether all values satisfy this check based on the presence of the exception.